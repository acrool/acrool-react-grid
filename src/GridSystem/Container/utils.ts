import {noXsMediaSizes, themeName} from '../../config';
import media from '../../media';
import {IContainerProps, NoXsMediaSize,TStyledProps} from '../../types';




/**
 * 產生 RWD Gutter 樣式
 * @param props
 */
export const createBreakpoint = (props: TStyledProps<IContainerProps>) => {
    const maxSizeConfig = getRWDMaxSize(props);
    return noXsMediaSizes.reduce((curr, sizeName) => {
        if(maxSizeConfig[sizeName]){
            return curr.concat(media[sizeName]`
                max-width: var(--acrool-container-max-width-${sizeName}, ${props.theme[themeName]?.containerMaxWidths[sizeName]}px);
            `);
        }
        return curr;
    }, []);
};


type TFluidConfig = Record<NoXsMediaSize, Record<NoXsMediaSize, boolean>>

/**
 * 產生 RWD MaxSize 樣式
 * @param props
 */
export const getRWDMaxSize = (props: TStyledProps<IContainerProps>) => {
    const fluidConfig: TFluidConfig = {
        sm: {sm: true, md: true, lg: true, xl: true, xxl: true},
        md: {sm: false, md: true, lg: true, xl: true, xxl: true},
        lg: {sm: false, md: false, lg: true, xl: true, xxl: true},
        xl: {sm: false, md: false, lg: false, xl: true, xxl: true},
        xxl: {sm: false, md: false, lg: false, xl: false, xxl: true},
    };

    if(typeof props.fluid !== 'undefined' && props.fluid !== false){
        if(props.fluid === true){
            return {sm: false, md: false, lg: false, xl: false, xxl: false};
        }
        return fluidConfig[props.fluid];
    }
    return {sm: true, md: true, lg: true, xl: true, xxl: true};
};





/**
 * 判斷是否為空
 * @param value
 */
const suffixCol = (value: any) => {
    if (!!Number(value)) {
        return `-${value}`;
    }
    if (value === 'auto') {
        return '-auto';
    }
    return '';
};

/**
 * 判斷是否為空
 * @param value
 */
const suffixContainer = (value: any) => {
    if (typeof value !== 'boolean') {
        return `-${value}`;
    }
    if(value){
        return '-fluid';
    }
    return '';
};

/**
 * 產生 Debug 資訊
 * @param props
 */
export const createInfo = (props: IContainerProps) => {
    return [
        props.fluid ? `container${suffixContainer(props.fluid)}`: 'container',
    ]
        .filter(Boolean)
        .join(' ');
};
