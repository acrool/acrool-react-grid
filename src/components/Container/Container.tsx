import React from 'react';
import styled, {css} from 'styled-components';
import media from '../../media';
import {mediaSizes, themeName} from '../../config';
import {FCChildrenProps, TStyledProps} from '../../types';



interface IProps extends FCChildrenProps{
    fluid?: boolean;
}


/**
 * 產生 Debug 資訊
 * @param props
 */
const generateDebugData = (props: TStyledProps<IProps>) => {
    if(process.env.NODE_ENV === 'production'){
        return undefined;
    }
    return [
        props.fluid ? 'container-fluid' : 'container',
    ]
        .filter(Boolean)
        .join(' ');
};


/**
 * 產生 RWD 樣式
 * @param props
 */
const generateRWDStyled = (props: TStyledProps<IProps>) => {
    return mediaSizes
        .map(sizeName => {
            return media[sizeName]`
            --bear-gutter-x: ${props.theme[themeName]?.gridGutterWidth}; 
            max-width: ${props.theme[themeName]?.containerMaxWidths[sizeName]}px;
        `;
        });
};








/**
 * 元件 Container
 */
const Container = styled.div.attrs((props: TStyledProps<IProps>) => ({
    'data-container': props.fluid ? 'fluid': '',
}))`
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
  padding-right: var(--bear-gutter-x);
  padding-left: var(--bear-gutter-x);
  
  ${(props: TStyledProps<IProps>) => css`
     --bear-gutter-x: ${props.theme[themeName]?.gridGutterWidth}; 

     ${!props.fluid && css`
        ${generateRWDStyled(props)};
    `}
 `}
`;

export default Container;
