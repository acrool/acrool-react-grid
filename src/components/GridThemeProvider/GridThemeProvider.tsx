import React, {PropsWithChildren} from 'react';
import {ThemeProvider, createGlobalStyle} from 'styled-components';

import {defaultGridTheme, themeName} from '../../config';
import {IGridSetting} from '../../types';


interface IProps {
    gridTheme: Partial<IGridSetting>;
    children: React.ReactNode;
}

const ThemeProviderFixed = ThemeProvider as unknown as React.FC<PropsWithChildren<{ theme: any }>>;


/**
 * Grid Theme Provider
 * use :
 *     <GridThemeProvider gridTheme={{
 *         gridGutterWidth: 5,
 *     }}>
 *          <Container>, <Row>, <Col> ...then eat this setting
 *     </GridThemeProvider>
 */
const GridThemeProvider = (props: IProps) => {
    const {gridTheme, children} = props;
    const {gridBreakpoints, containerMaxWidths, gridGutterWidthMedia, ...baseDefaultTheme} = defaultGridTheme;

    // console.log('test render bear-react-grid version 10');
    const composeGridTheme = {
        [themeName]: {
            ...baseDefaultTheme,
            ...gridTheme,
            gridBreakpoints: Object.assign({}, gridBreakpoints, gridTheme?.gridBreakpoints),
            containerMaxWidths: Object.assign({}, containerMaxWidths, gridTheme?.containerMaxWidths),
            gridGutterWidthMedia: Object.assign({}, gridGutterWidthMedia, gridTheme?.gridGutterWidthMedia),
        }
    };

    return <ThemeProviderFixed theme={composeGridTheme}>
        {children}
        <GlobalStyled/>
    </ThemeProviderFixed>;
};

export default GridThemeProvider;


const GlobalStyled = createGlobalStyle`
    
`;
