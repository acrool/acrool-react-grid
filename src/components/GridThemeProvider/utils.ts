import {TStyledProps, IRowProps, NoXsMediaSize} from '../../types';
import {mediaSizes} from '../../config';
import media from '../../media';


const renderMarginStyle = (sizeName: NoXsMediaSize) => {
    return Array.from({length: 6}).map((row,idx) => {
        return `
        .m-${sizeName}-${idx} {margin: var(--bear-spacer-${idx}) !important;}
        .mt-${sizeName}-${idx}, .my-${sizeName}-${idx} {margin-top: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        .mr-${sizeName}-${idx}, .mx-${sizeName}-${idx} {margin-right: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        .mb-${sizeName}-${idx}, .my-${sizeName}-${idx} {margin-bottom: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        .ml-${sizeName}-${idx}, .mx-${sizeName}-${idx} {margin-left: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        `;
    }).join('');
};

const renderPaddingStyle = (sizeName: NoXsMediaSize) => {
    return Array.from({length: 6}).map((row,idx) => {
        return `
        .p-${sizeName}-${idx} {margin: var(--bear-spacer-${idx}) !important;}
        .pt-${sizeName}-${idx}, .py-${sizeName}-${idx} {padding-top: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        .pr-${sizeName}-${idx}, .px-${sizeName}-${idx} {padding-right: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        .pb-${sizeName}-${idx}, .py-${sizeName}-${idx} {padding-bottom: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        .pl-${sizeName}-${idx}, .px-${sizeName}-${idx} {padding-left: ${idx === 0 ? 0: `var(--bear-spacer-${idx})`} !important;}
        `;
    }).join('');
};

const renderOrderStyle = (sizeName: NoXsMediaSize) => {
    return Array.from({length: 6}).map((row,idx) => {
        return `
        .order-${sizeName}-${idx} {order: ${idx}!important;}
        `;
    }).join('');
};


/**
 * 產生 RWD 樣式
 * @param props
 */
export const generateRWDStyled = (props: TStyledProps<IRowProps>) => {
    return mediaSizes
        .map(sizeName => {
            return media[sizeName]`
            .d-${sizeName}-none {display: none !important;}
            .d-${sizeName}-inline {display: inline !important;}
            .d-${sizeName}-inline-block {display: inline-block !important;}
            .d-${sizeName}-block {display: block !important;}
            .d-${sizeName}-flex {display: flex !important;}
            .d-${sizeName}-inline-flex {display: inline-flex !important;}
            .d-${sizeName}-table-cell {display: table-cell !important;}
            .d-${sizeName}-table {display: table !important;}
            .d-${sizeName}-table-row {display: table-row !important;}

            .flex-${sizeName}-row {flex-direction: row !important;}
            .flex-${sizeName}-column {flex-direction: column !important;}
            .flex-${sizeName}-row-reverse {flex-direction: row-reverse !important;}
            .flex-${sizeName}-column-reverse {flex-direction: column-reverse !important;}

            .flex-${sizeName}-wrap {flex-wrap: wrap !important;}
            .flex-${sizeName}-nowrap {flex-nowrap: nowrap !important;}
            .flex-${sizeName}-wrap-reverse {flex-nowrap: wrap-reverse !important;}

            .flex-${sizeName}-fill {flex: 1 1 auto !important;}
            .flex-${sizeName}-grow-0 {flex-grow: 0 !important;}
            .flex-${sizeName}-grow-1 {flex-grow: 1 !important;}
            .flex-${sizeName}-shrink-0 {flex-shrink: 0 !important;}
            .flex-${sizeName}-shrink-1 {flex-shrink: 1 !important;}

            .justify-content-${sizeName}-start {justify-content: flex-start !important;}
            .justify-content-${sizeName}-end {justify-content: flex-end !important;}
            .justify-content-${sizeName}-center {justify-content: flex-center !important;}
            .justify-content-${sizeName}-between {justify-content: space-between !important;}
            .justify-content-${sizeName}-around {justify-content: space-around !important;}

            .align-items-${sizeName}-start {align-items: flex-start !important;}
            .align-items-${sizeName}-end {align-items: flex-end !important;}
            .align-items-${sizeName}-center {align-items: center !important;}
            .align-items-${sizeName}-baseline {align-items: baseline !important;}
            .align-items-${sizeName}-stretch {align-items: stretch !important;}

            .align-content-${sizeName}-start {align-content: start !important;}
            .align-content-${sizeName}-end {align-content: flex-end !important;}
            .align-content-${sizeName}-center {align-content: center !important;}
            .align-content-${sizeName}-between {align-content: between !important;}
            .align-content-${sizeName}-around {align-content: around !important;}
            .align-content-${sizeName}-stretch {align-content: stretch !important;}

            .align-self-${sizeName}-start {align-self: flex-start !important;}
            .align-self-${sizeName}-end {align-self: flex-end !important;}
            .align-self-${sizeName}-center {align-self: center !important;}
            .align-self-${sizeName}-baseline {align-self: baseline !important;}
            .align-self-${sizeName}-stretch {align-self: stretch !important;}

            .text-${sizeName}-justify {text-align: justify !important;}
            .text-${sizeName}-left {text-align: left !important;}
            .text-${sizeName}-right {text-align: right !important;}
            .text-${sizeName}-center {text-align: center !important;}

            .m-${sizeName}-auto {margin: var(--bear-spacer-auto) !important;}
            .mt-${sizeName}-auto, .my-${sizeName}-auto {margin-top: auto !important;}
            .mr-${sizeName}-auto, .mx-${sizeName}-auto {margin-right: auto !important;}
            .mb-${sizeName}-auto, .my-${sizeName}-auto {margin-bottom: auto !important;}
            .ml-${sizeName}-auto, .mx-${sizeName}-auto {margin-left: auto !important;}

            ${renderMarginStyle(sizeName)}
            ${renderPaddingStyle(sizeName)}
            ${renderOrderStyle(sizeName)}

            .overflow-x-${sizeName}-hidden {overflow-x: hidden!important;}
            .overflow-y-${sizeName}-hidden {overflow-y: hidden!important;}
            .overflow-x-${sizeName}-auto {overflow-x: auto!important;}
            .overflow-y-${sizeName}-auto {overflow-y: auto!important;}
        `;
        });
};

