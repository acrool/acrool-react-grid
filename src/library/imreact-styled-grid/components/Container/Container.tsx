import React from 'react';
import styled, {css} from 'styled-components/macro';
import {ContainerProps} from './types';

import {themeName} from '../../config';
import media from '../../media';
import getDataName from './getDataName';

const generateMedia = (props: any) => Object.keys(props.theme.imreactStyledGrid.gridBreakpoints)
    .map(sizeName => {
        if (props.theme.imreactStyledGrid.gridBreakpoints[sizeName] > 0) {
            return media[sizeName]`
                max-width: ${props.theme.imreactStyledGrid.containerMaxWidths[sizeName]}px;
            `;
        }
        return false;
    })
    .filter(row => row !== false);
;

/**
 * Row Component
 */
const Container: any = styled.div.attrs((props: ContainerProps) => ({
    'data-grid': 'container',
    'data-debug': getDataName(props),
}))`
  width: 100%;
  margin-right: auto;
  margin-left: auto;

  ${(props: any) => css`
     padding-right: ${props.theme.imreactStyledGrid?.gridGutterWidth}px;
     padding-left: ${props.theme.imreactStyledGrid?.gridGutterWidth}px;

     ${!props.fluid && css`
        ${generateMedia(props)};
    `}
 `}
`;

export default (props: ContainerProps) => <Container {...props}/>;
