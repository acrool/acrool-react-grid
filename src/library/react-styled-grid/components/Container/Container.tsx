import React from 'react';
import styled, {css} from 'styled-components/macro';
import {ContainerProps} from './types';

import {themeName} from '../../config';
import media from '../../media';
import getDataName from './getDataName';

const generateMedia = (props: any) => Object.keys(props.theme.reactStyledGrid.gridBreakpoints)
    .map(sizeName => {
        if (props.theme.reactStyledGrid.gridBreakpoints[sizeName] > 0) {
            return media[sizeName]`
                max-width: ${props.theme.reactStyledGrid.containerMaxWidths[sizeName]}px;
            `;
        }
        return false;
    })
    .filter(row => row !== false);
;

/**
 * Row Component
 */
const Container: any = styled.div.attrs((props: ContainerProps) => ({
    'data-grid': 'container',
    'data-debug': getDataName(props),
}))`
  width: 100%;
  margin-right: auto;
  margin-left: auto;

  ${(props: any) => css`
     padding-right: ${props.theme.reactStyledGrid?.gridGutterWidth}px;
     padding-left: ${props.theme.reactStyledGrid?.gridGutterWidth}px;

     ${!props.fluid && css`
        ${generateMedia(props)};
    `}
 `}
`;

export default (props: ContainerProps) => <Container {...props}/>;
