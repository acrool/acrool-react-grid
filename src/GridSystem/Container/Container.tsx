import styled, {css} from 'styled-components';
import {themeName} from '../../config';
import {TStyledProps, IContainerProps} from '../../types';
import {createBreakpoint} from './utils';
import {cssGetter} from '../Col/utils';


/**
 * 判斷是否為空
 * @param value
 */
const suffixCol = (value: any) => {
    if (!!Number(value)) {
        return `-${value}`;
    }
    if (value === 'auto') {
        return '-auto';
    }
    return '';
};

/**
 * 判斷是否為空
 * @param value
 */
const suffixContainer = (value: any) => {
    if (!!Number(value)) {
        return `-${value}`;
    }
    return '';
};

/**
 * 產生 Debug 資訊
 * @param props
 */
const generateDebugData = (props: TStyledProps<IContainerProps>) => {
    return [
        'container',
        props.fluid && `container-fluid${suffixContainer(props.fluid)}`,
        props.col && `col${suffixCol(props.col)}`,
        props.sm && `col-sm${suffixCol(props.sm)}`,
        props.md && `col-md${suffixCol(props.md)}`,
        props.lg && `col-lg${suffixCol(props.lg)}`,
        props.xl && `col-xl${suffixCol(props.xl)}`,
        props.xxl && `col-xxl${suffixCol(props.xxl)}`,
    ]
        .filter(Boolean)
        .join(' ');
};


/**
 * Container Component
 */
const Container = styled.div.attrs((props: TStyledProps<IContainerProps>) => ({
    'data-grid': generateDebugData(props),
}))`
  --bear-gutter-x: ${props => props.theme[themeName]?.gutter};
  --bear-gutter-y: 0;

  width: 100%;
  margin-right: auto;
  margin-left: auto;
  padding-right: var(--bear-gutter-x);
  padding-left: var(--bear-gutter-x);

  ${(props: TStyledProps<IContainerProps>) => css`

      ${props.fluid && css`
         max-width: none;
     `}
      ${props.col && cssGetter.col(props.col, props.theme[themeName]?.gridColumns)};

      ${createBreakpoint(props)};

  `}
`;

export default Container;
