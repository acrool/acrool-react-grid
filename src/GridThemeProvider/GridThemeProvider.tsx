import React, {PropsWithChildren} from 'react';
import {ThemeProvider, createGlobalStyle, css} from 'styled-components';

import {defaultGridTheme, themeName} from '../config';
import {TStyledProps, TThemeProps, IGridThemeProviderProps} from '../types';
import {generateRWDStyled} from './utils';
import '../styles.css';



const ThemeProviderFixed = ThemeProvider as unknown as React.FC<PropsWithChildren<{ theme: any }>>;


/**
 * Grid Theme Provider
 * use :
 *     <GridThemeProvider gridTheme={{
 *         gridGutterWidth: 5,
 *     }}>
 *          <Container>, <Row>, <Col> ...then eat this setting
 *     </GridThemeProvider>
 */
const GridThemeProvider = (props: IGridThemeProviderProps) => {
    const {gridTheme, children} = props;
    const {gridBreakpoints, containerMaxWidths, gridGutterWidthMedia, ...baseDefaultTheme} = defaultGridTheme;

    // console.log('test render bear-react-grid version 10');
    const composeGridTheme = {
        [themeName]: {
            ...baseDefaultTheme,
            ...gridTheme,
            gridBreakpoints: Object.assign({}, gridBreakpoints, gridTheme?.gridBreakpoints),
            containerMaxWidths: Object.assign({}, containerMaxWidths, gridTheme?.containerMaxWidths),
            gridGutterWidthMedia: Object.assign({}, gridGutterWidthMedia, gridTheme?.gridGutterWidthMedia),
        }
    };

    return <ThemeProviderFixed theme={composeGridTheme}>
        {children}
        <GlobalStyled/>
    </ThemeProviderFixed>;
};

export default GridThemeProvider;


const GlobalStyled = createGlobalStyle`
    :root{
        --bear-spacer: 1rem;
        --bear-spacer-1: calc(1rem * .25);
        --bear-spacer-2: calc(1rem * .5);
        --bear-spacer-3: calc(1rem);
        --bear-spacer-4: calc(1rem * 1.5);
        --bear-spacer-5: calc(1rem * 3);
    }

    ${(props: TStyledProps<TThemeProps>) => css`
        ${generateRWDStyled(props)}
    `}

    [data-row][data-horizontal="left"]{
        justify-content: flex-start;
    }
    [data-row][data-horizontal="center"]{
        justify-content: center;
    }
    [data-row][data-horizontal="right"]{
        justify-content: flex-end;
    }
    [data-row][data-vertical="top"]{
        justify-content: flex-start;
    }
    [data-row][data-vertical="center"]{
        justify-content: center;
    }
    [data-row][data-vertical="bottom"]{
        justify-content: flex-end;
    }
`;
