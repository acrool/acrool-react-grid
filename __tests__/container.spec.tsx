import React from 'react';
import 'jest-styled-components';
import {render} from '@testing-library/react';
import '@testing-library/dom';
import GridThemeProvider from '../src/GridThemeProvider';
import Container from '../src/GridSystem/Container';



describe('Container', () => {
    it('renders container', () => {
        const {container} = render(<Container>container</Container>, {wrapper: GridThemeProvider});
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders container fluid', () => {
        const {container} = render(<Container fluid>container fluid</Container>, {wrapper: GridThemeProvider});
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders container fluid="md"', () => {
        const {container} = render(<Container fluid="md">container-md</Container>, {wrapper: GridThemeProvider});
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders container fluid="lg"', () => {
        const {container} = render(<Container fluid="lg">container-lg</Container>, {wrapper: GridThemeProvider});
        expect(container.firstChild).toMatchSnapshot();
    });
});
