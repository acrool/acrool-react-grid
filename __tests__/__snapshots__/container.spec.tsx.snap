// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders container 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  padding-right: var(--acrool-gutter-x, 0.75rem);
  padding-left: var(--acrool-gutter-x, 0.75rem);
}

@media (min-width: 576px) {
  .c0 {
    max-width: var(--acrool-container-max-width-sm, 540px);
  }
}

@media (min-width: 768px) {
  .c0 {
    max-width: var(--acrool-container-max-width-md, 720px);
  }
}

@media (min-width: 992px) {
  .c0 {
    max-width: var(--acrool-container-max-width-lg, 960px);
  }
}

@media (min-width: 1200px) {
  .c0 {
    max-width: var(--acrool-container-max-width-xl, 1140px);
  }
}

@media (min-width: 1540px) {
  .c0 {
    max-width: var(--acrool-container-max-width-xxl, 1540px);
  }
}

<div
  class="c0"
  data-grid="container"
>
  container
</div>
`;

exports[`Container renders container fluid 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  padding-right: var(--acrool-gutter-x, 0.75rem);
  padding-left: var(--acrool-gutter-x, 0.75rem);
}

<div
  class="c0"
  data-grid="container-fluid"
>
  container fluid
</div>
`;

exports[`Container renders container fluid="lg" 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  padding-right: var(--acrool-gutter-x, 0.75rem);
  padding-left: var(--acrool-gutter-x, 0.75rem);
}

@media (min-width: 992px) {
  .c0 {
    max-width: var(--acrool-container-max-width-lg, 960px);
  }
}

@media (min-width: 1200px) {
  .c0 {
    max-width: var(--acrool-container-max-width-xl, 1140px);
  }
}

@media (min-width: 1540px) {
  .c0 {
    max-width: var(--acrool-container-max-width-xxl, 1540px);
  }
}

<div
  class="c0"
  data-grid="container-lg"
>
  container-lg
</div>
`;

exports[`Container renders container fluid="md" 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  padding-right: var(--acrool-gutter-x, 0.75rem);
  padding-left: var(--acrool-gutter-x, 0.75rem);
}

@media (min-width: 768px) {
  .c0 {
    max-width: var(--acrool-container-max-width-md, 720px);
  }
}

@media (min-width: 992px) {
  .c0 {
    max-width: var(--acrool-container-max-width-lg, 960px);
  }
}

@media (min-width: 1200px) {
  .c0 {
    max-width: var(--acrool-container-max-width-xl, 1140px);
  }
}

@media (min-width: 1540px) {
  .c0 {
    max-width: var(--acrool-container-max-width-xxl, 1540px);
  }
}

<div
  class="c0"
  data-grid="container-md"
>
  container-md
</div>
`;
